File: Museum/Frontend/museum-frontend-angular/src/app/models/MuseumDTO.ts
import { MuseumType } from "./MuseumType";

export interface MuseumDTO{
    museumID: number;
    name: string;
    address: string;
    phone: string;
    city: string;
    country: string;
    longitude: number;
    latitude: number;
    thumbnail: string;
    type: MuseumType;
}

File: Museum/Backend/REST service/src/main/java/museum/service/models/DTOs/MuseumDTO.java
package museum.service.models.DTOs;

import lombok.Data;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Data
public class MuseumDTO
{
    private Integer museumID;

    @NotBlank
    private String name, address, phone, city, country;

    @NotBlank
    private String type;

    @NotNull
    private BigDecimal longitude, latitude;
	
	private String thumbnail;
	
	private MuseumTypeDTO type;
}
